core/controllers/questions_list.py

```diff
@@ -102,17 +103,16 @@ class QuestionCountDataHandler(base.BaseHandler):
     def get(self, comma_separated_skill_ids):
         """Handles GET requests."""
         skill_ids = comma_separated_skill_ids.split(',')
+        skill_ids = list(set(skill_ids))
 
-        for skill_id in skill_ids:
-            try:
-                skill_domain.Skill.require_valid_skill_id(skill_id)
-            except Exception:
-                raise self.PageNotFoundException(Exception('Invalid skill id'))
+        try:
+            skill_domain.Skill.require_valid_skill_ids(skill_ids)
+        except utils.ValidationError:
+            raise self.InvalidInputException('Invalid skill id')
 
-        question_count = question_services.get_question_count_by_skill_ids(
-            skill_ids)
+        total_question_count = (
+            question_services.get_question_count_for_skill_ids(skill_ids))
 
-        self.values.update({
-            'question_count': question_count
+        return self.render_json({
```

Comment:
This isn't the usual way it's handled, right? (We don't usually return the result of self.render_json().) Please ensure this matches what is done in other handlers.

Reply:
Yes fixed, I noticed that with this method, no extra stuff is sent back, like username, or is_admin, and hence used this.
Anyways, fixed.